src/util/merkleblock.rs

```diff
@@ -420,32 +421,40 @@ impl MerkleBlock {
     /// // Create a merkle block containing a single transaction
     /// let txid = Txid::from_hex(
     ///     "5a4ebf66822b0b2d56bd9dc64ece0bc38ee7844a23ff1d7320a88c5fdb2ad3e2").unwrap();
-    /// let match_txids = vec![txid].into_iter().collect();
-    /// let mb = MerkleBlock::from_block(&block, &match_txids);
+    /// let match_txids: Vec<Txid> = vec![txid].into_iter().collect();
+    /// let mb = MerkleBlock::from_block_p(&block, |t| match_txids.contains(t));
     ///
     /// // Authenticate and extract matched transaction ids
     /// let mut matches: Vec<Txid> = vec![];
     /// let mut index: Vec<u32> = vec![];
     /// assert!(mb.extract_matches(&mut matches, &mut index).is_ok());
     /// assert_eq!(txid, matches[0]);
     /// ```
-    pub fn from_block(block: &Block, match_txids: &HashSet<Txid>) -> Self {
+    pub fn from_block_p<F>(block: &Block, match_txids: F) -> Self
```

Comment:
what's the `p` stands for?

Reply:
+1. I read it as "predicate" which implies that these should return booleans, but they don't.

If this rename was merely to avoid an API break, I think we should just let the API break happen (we are already API-breaking in this PR by breaking `--no-default-features` if nothing else).

Reply:
It's also the kind of API break permitted by Rust itself so it'd seem a bit strange to be more strict about API stability.

Reply:
Right, I understood that we could avoid the major version bump by only breaking `--no-default-features`, apparently that is not the case, so we should just break the API here.

Reply:
Ah, hmm, I may be in the wrong here. If this is really the *only* thing preventing us from being API-compatible (in the Rust sense) then I retract my complaint.

Although I still don't like the `_p` suffix.

Reply:
> Right, I understood that we could avoid the major version bump by only breaking `--no-default-features`, apparently that is not the case, so we should just break the API here.

The API *is* kept stable - see old signature right below.

I changed the new signature to be less cryptic.